traffic_ops/traffic_ops_golang/parameter/parameters.go

```diff
@@ -259,3 +264,351 @@ func deleteQuery() string {
+		ValueQueryParam:      {Column: "p.value"},
+	}
+	if _, ok := inf.Params["orderby"]; !ok {
+		inf.Params["orderby"] = "name"
+	}
+	where, orderBy, pagination, queryValues, errs := dbhelpers.BuildWhereAndOrderByAndPagination(inf.Params, queryParamsToQueryCols)
+	if len(errs) > 0 {
+		api.HandleErr(w, r, tx.Tx, http.StatusBadRequest, util.JoinErrs(errs), nil)
+		return
+	}
+
+	if inf.Config.UseIMS {
+		runSecond, maxTime = ims.TryIfModifiedSinceQuery(tx, r.Header, queryValues, selectMaxLastUpdatedQuery(where))
+		if !runSecond {
+			log.Debugln("IMS HIT")
+			api.AddLastModifiedHdr(w, maxTime)
+			w.WriteHeader(http.StatusNotModified)
+			return
+		}
+		log.Debugln("IMS MISS")
+	} else {
+		log.Debugln("Non IMS request")
+	}
+
+	query := selectQuery() + where + ParametersGroupBy() + orderBy + pagination
+	rows, err := tx.NamedQuery(query, queryValues)
+	if err != nil {
+		api.HandleErr(w, r, tx.Tx, http.StatusInternalServerError, nil, fmt.Errorf("Parameter read: error getting Parameter(s): %w", err))
+		return
+	}
+	defer log.Close(rows, "unable to close DB connection")
+
+	params := tc.ParameterNullableV5{}
+	paramsList := []tc.ParameterNullableV5{}
+	for rows.Next() {
+		if err = rows.Scan(&params.ConfigFile, &params.ID, &params.LastUpdated, &params.Name, &params.Value, &params.Secure, &params.Profiles); err != nil {
+			api.HandleErr(w, r, tx.Tx, http.StatusInternalServerError, nil, fmt.Errorf("error getting parameter(s): %w", err))
+			return
+		}
+		if params.Secure != nil && *params.Secure {
+			if inf.Version.Major >= 4 &&
+				inf.Config.RoleBasedPermissions &&
+				!inf.User.Can("PARAMETER-SECURE:READ") {
+				params.Value = &HiddenField
+			} else if inf.User.PrivLevel < auth.PrivLevelAdmin {
+				params.Value = &HiddenField
+			}
+		}
+
+		paramsList = append(paramsList, params)
+	}
+
+	api.WriteResp(w, r, paramsList)
+	return
+}
+
+func CreateParameter(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+	tx := inf.Tx.Tx
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, errors.New("error reading request body"), nil)
+		return
+	}
+	defer r.Body.Close()
+
+	// Initial Unmarshal to validate request body
+	var data interface{}
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, errors.New("invalid request format"), nil)
+		return
+	}
+
+	// This code block decides if the request body is a slice of parameters or a single object.
+	var params []tc.ParameterV5
+	switch reflect.TypeOf(data).Kind() {
+	case reflect.Slice:
+		if err := json.Unmarshal(body, &params); err != nil {
+			api.HandleErr(w, r, tx, http.StatusBadRequest, errors.New("error unmarshalling slice"), nil)
+			return
+		}
+	case reflect.Map:
```

Comment:
Reflection is slow, using it is a no-go. Instead of reflecting, try typecasting:
```go
if my_array, ok := data.([]tc.ParameterV5); ok {
	/* ... */
} else if my_parameter, ok := data.(tc.ParameterV5); ok {
	/* ... */
} else {
	/* ... */
}
```

Reply:
(Reflection in tests is fine, of course, I just mean reflection should be avoided in production code)

Reply:
I understand. Removed the usage of reflection.